### DISCUSSION TITLE
Is it possible to add more sections in Simplemenu? And some other questions.

### DISCUSSION THREAD

**Discussion by nafields** [rg35xx-cfw/Koriki#51] [Q&A] [answered]
First and foremost I'm new to this group, and new to koriki. I love it (minus the pre-alpha state bugs and blah blah blah) >> It's great honestly. 

**My first question:** _Is it possible to add more sections in simplemenu than the basic App and Games, Consoles, Handhelds, etc.?_
To test drive this:

- [x] I added a new  script in the sections folder with the appropriate script.
- [x] Created an ini with the correct console names, links, etc. (not sure if there can be newlines at the end before EOF. (I read they are very picky on the commas in their extensions list.) in the section_groups folder
- [x]  Added a base default .png to that folder
- [x] I even added a picture in the sections_groups section of the theme, even though it should use default text if missing.

Curious if this is even possible, or if I've just overlooked something simple. The theme ini file and the simplemenu ini file don't appear to have anything in them related to this. I did not poke around in the BATOCERA partition to see if there were settings in there or not, but I'm fairly certain it's just a swap disk and embedded code to make this run.

I'm also curious if it's possible to add a script that doesn't go back to the rom / game selection menu, but goes back to the main menu on reboot or wake from sleep. I believe this is probably in one of the save state files, but I didn't have time to debug what the values being saved are, and thought it better just to ask here.

I could then make some of these settings/configurations and some other things I've been playing with available as an "app" in the menus to turn on or off just using a shell script.

 More than happy to help in anyway shape or form. I do program, just not embedded and generally not for linux.

 P.S. I'm in the process of porting over a theme based on epic-noir, if your interested in it. I know that the base themes are just .pngs right now, but I already built an internal app to port any emulation station themes over with little setup, assuming they are xml based. (needs some work before public). Let me know if you foresee simplemenu ever being usable in that regard on the rg35xx.

Great work all around. Appreciate this little device.

Cheers

**Comment by acmeplus** [comment, marked as answer]
Yes, it's possible to add more sections to simplemenu, the only thing you need to do is to add a new sectionX.ini to system/.simplemenu/section_groups. I recommend you use the consoles.ini as a template. Remember to keep the CONSOLES keyword at the beginning, since that's what simplemenu looks for. Then you can define there any system that you want and in each system define the path, launcher, and extension.

About going back to the root menu, it's not currently part of the core functionally of simplemenu, but I'll consider that for simplermenu+.

About porting other themes, yes, any contributions are welcomed. For simplermenu+ the goal is to eventually support not just simplemenu themes but also emulationstation ones. Also to be able to define themes dynamically with text strings and not just graphics.
